ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableProcessorTest.kt

```diff
@@ -13,45 +17,117 @@ import java.util.Locale
+            override fun getApplicationName(): String {
+                return "test app name"
+            }
+        }
+        variableDataProvider = VariableDataProvider(applicationContext)
+        rcPackage = TestData.Packages.annual
+    }
+
+    @Test
+    fun `process variables returns original text if empty string`() {
+        val originalText = ""
+        expectVariablesResult(originalText, originalText)
     }
 
     @Test
     fun `process variables returns original text if no variables`() {
         val originalText = "text without any variables"
-        val resultText = VariableProcessor.processVariables(variableDataProvider, originalText, rcPackage, locale)
-        assertThat(resultText).isEqualTo(originalText)
+        expectVariablesResult(originalText, originalText)
     }
 
     @Test
     fun `process variables returns processed text with single variable`() {
         val originalText = "text with {{ app_name }} one variable"
-        every { variableDataProvider.applicationName } returns "app name"
-        val resultText = VariableProcessor.processVariables(variableDataProvider, originalText, rcPackage, locale)
-        assertThat(resultText).isEqualTo("text with app name one variable")
+        val expectedText = "text with test app name one variable"
+        expectVariablesResult(originalText, expectedText)
     }
 
     @Test
     fun `process variables returns processed text with multiple variable`() {
         val originalText = "text with {{ app_name }} and {{ sub_price_per_month }} multiple variables"
-        every { variableDataProvider.applicationName } returns "app name"
-        every { variableDataProvider.localizedPricePerMonth(rcPackage, locale) } returns "$9.99"
-        val resultText = VariableProcessor.processVariables(variableDataProvider, originalText, rcPackage, locale)
-        assertThat(resultText).isEqualTo("text with app name and $9.99 multiple variables")
+        val expectedText = "text with test app name and $5.67 multiple variables"
+        expectVariablesResult(originalText, expectedText)
     }
 
     @Test
     fun `process variables does not modify unknown variables`() {
         val originalText = "text with {{ unknown_variable }}"
+        expectVariablesResult(originalText, originalText)
+    }
+
+    @Test
+    fun `process variables does not process variable if no spaces`() {
+        val originalText = "text with {{app_name}} and something"
+        expectVariablesResult(originalText, originalText)
+    }
+
+    // region Variables
+
+    @Test
+    fun `process variables processes app_name`() {
+        expectVariablesResult("{{ app_name }}", "test app name")
+    }
+
+    @Test
+    fun `process variables processes price`() {
+        expectVariablesResult("{{ price }}", "$67.99")
+    }
+
+    @Test
+    fun `process variables processes price_per_period`() {
+        expectVariablesResult("{{ price_per_period }}", "PRICE_PER_PERIOD")
```

Comment:
Using the mock data for now. Once we have an implementation we can just update the tests.

Reply:
👍🏻